### DISCUSSION TITLE
skywalking may contain stored xss problem

### DISCUSSION THREAD

**Discussion by creeew** [apache/skywalking#8243] [Q&A] [answered]
skywalking maybe vulnerable if received a trace report with xss script. Like in logs contain xss script or service name is a xss script text.

**Comment by wu-sheng** [comment, marked as answer]
Don't open a trace report to the internet, generally. We provided browser monitoring, but about this kind of data report, you should build you own validating mechanism before any data into the Database. We are open source, so the data format is open to you already.

**Reply by creeew** [reply]
Appreciate it, I gonna replace xss vulnerable characters in zull post filter to solve this problem.

**Reply by wu-sheng** [reply]
If you want to share your experience to the community(don't have to), you could submit a blog to the website sharing what you have done.
